bson/_cbsonmodule.c

```diff
@@ -2797,6 +2816,85 @@ static PyObject* _cbson_decode_all(PyObject* self, PyObject* args) {
     return result;
 }
 
+
+static PyObject* _cbson_array_of_documents_to_buffer(PyObject* self, PyObject* args) {
+    uint32_t size;
+    uint32_t value_length;
+    uint32_t position = 0;
+    buffer_t buffer;
+    const char* string;
+    PyObject* arr;
+    PyObject* result = NULL;
+    Py_buffer view = {0};
+
+    if (!PyArg_ParseTuple(args, "O", &arr)) {
+        return NULL;
+    }
+
+    buffer = pymongo_buffer_new();
+
+    if (!_get_buffer(arr, &view)) {
+        goto fail;
+    }
+
+    string = (char*)view.buf;
+    memcpy(&size, string, 4);
+    size = BSON_UINT32_FROM_LE(size);
+
+    if (view.len < size || view.len > BSON_MAX_SIZE) {
+        PyObject* InvalidBSON = _error("InvalidBSON");
+        if (InvalidBSON) {
+            PyErr_SetString(InvalidBSON, "objsize too large");
+            Py_DECREF(InvalidBSON);
+        }
+        goto done;
+    }
+
+    position += 4;
+    while (position < size - 1) {
+        // Just skip the keys.
+        while (string[position] != '\0') {
+            position++;
+        }
+        position++;
+
+        memcpy(&value_length, string + position, 4);
+        value_length = BSON_UINT32_FROM_LE(value_length);
+
+        if (pymongo_buffer_write(buffer, string + position, value_length) == 1) {
+            PyObject* InvalidBSON = _error("InvalidBSON");
```

Comment:
An exception should already be set in this case (it would be a MemoryError because the buffer could not be resized) so we don't need this InvalidBSON logic.

Reply:
Removed

Reply:
Removed